app/gui/src/dashboard/layouts/VersionChecker.tsx

```diff
@@ -1,88 +1,204 @@
 
+import { Stepper } from '#/components/Stepper'
+import { useEventCallback } from '#/hooks/eventCallbackHooks'
 import { download } from '#/utilities/download'
 import { getDownloadUrl, getLatestRelease, LATEST_RELEASE_PAGE_URL } from '#/utilities/github'
+import { startTransition, useState } from 'react'
 
-// ======================
-// === VersionChecker ===
-// ======================
+const CURRENT_VERSION = process.env.ENSO_CLOUD_DASHBOARD_VERSION
+// eslint-disable-next-line @typescript-eslint/no-magic-numbers
+const STALE_TIME = 24 * 60 * 60 * 1000 // 1 day
+// eslint-disable-next-line @typescript-eslint/no-magic-numbers
+const STALE_TIME_ERROR = 10 * 60 * 1000 // 10 minutes
 
 /** Check the version. */
 export default function VersionChecker() {
-  const [isOpen, setIsOpen] = React.useState(false)
-  const { getText } = useText()
+  const [isOpen, setIsOpen] = useState(false)
+
+  const { getText, locale } = useText()
   const toastAndLog = useToastAndLog()
   const localBackend = useLocalBackend()
+
   const supportsLocalBackend = localBackend != null
-  const enableVersionChecker = useEnableVersionChecker() ?? (!IS_DEV_MODE && supportsLocalBackend)
+  const overrideValue = useEnableVersionChecker()
+  const setOverrideValue = useSetEnableVersionChecker()
 
+  const enableVersionChecker = (() => {
+    // In dev mode we can override the behavior using the devtools.
+    if (IS_DEV_MODE) {
+      return overrideValue ?? false
+    }
+
+    // Otherwise, we only check the version if we're connected to the local backend,
+    // because usually that means we're running inside an electron app.
+    return supportsLocalBackend
+  })()
+
+  const queryClient = useQueryClient()
   const metadataQuery = useQuery({
-    queryKey: ['latestRelease', enableVersionChecker],
-    queryFn: () => (enableVersionChecker ? getLatestRelease() : null),
+    queryKey: ['latestRelease'],
+    queryFn: () => getLatestRelease(),
+    select: (data) => ({
+      tagName: data.tag_name,
+      publishedAt: data.published_at,
+      htmlUrl: data.html_url,
+    }),
+    enabled: enableVersionChecker,
+    meta: { persist: false },
+    staleTime: (query) => {
+      if (query.state.error) {
+        return STALE_TIME_ERROR
+      }
+
+      return STALE_TIME
+    },
   })
-  const latestVersion = metadataQuery.data?.tag_name
 
-  React.useEffect(() => {
-    if (latestVersion != null && latestVersion !== process.env.ENSO_CLOUD_DASHBOARD_VERSION) {
-      setIsOpen(true)
+  const { stepperState, isLastStep, resetStepper } = Stepper.useStepperState({ steps: 2 })
+
+  const remindLater = useEventCallback(() => {
+    setIsOpen(false)
+    // We need to suppress the dialog from showing again for next 24 hours.
+    queryClient.setQueryData(['latestRelease'], {
+      /* eslint-disable @typescript-eslint/naming-convention, camelcase */
+      tag_name: CURRENT_VERSION,
+      published_at: new Date().toISOString(),
```

Comment:
nit: we typically use `toRfc3339` although it's only *really* needed to convert it to the newtype used by the backend APIs - although we may want to use it here anyway just for consistency.

Reply:
@somebody1234 gh returns date in ISO8601, though I think the form is comatible with the RFC3339 in that case, eg: "2024-10-11T12:43:56Z". I think that the format here should be more compatible with GH api instead? Though at this point it doesn't matter as it doesn't affect anything

Reply:
yeah i believe the backend calls it rfc 3339. but yeah `toISOString()` is probably fine too